raiden/network/proxies/discovery.py

```diff
@@ -49,7 +49,7 @@ def __init__(
             'latest',
         )
 
-        if result == '0x':
+        if result == b'0x':
```

Comment:
> Do we really need to add this b before all of them?

The `b` prefix changes the type of the object, in python3 `b''` is a binary string and simple quotes `''` is the same as `u''`, a unicode string.

```python
>>> type(b'')
<class 'bytes'>
>>> type('')
<class 'str'>
>>> type(u'')
<class 'str'>
>>> b'' == u''
False
```

Reply:
so string literals in python are not looking at the file encoding but are always going to be utf8 in python 3? I see.

Is there no way at all to enforce binary string everywhere?

Reply:
> are always going to be utf8 in python 3?

Actually it's *unicode*, the underlying representation is implementation specific and unimportant.

> Is there no way at all to enforce binary string everywhere?

Not that I'm aware of, and why would you want to do this?

Reply:
So that we don't have to write `b'blablabla'` all the time if we always want that encoding.

Reply:
> Actually it's unicode, the underlying representation is implementation specific and unimportant.

I see. So in order to access any bytes at all you have to go down to a bytestring which would then force you to think in terms of bytes and encodings.

But as long as you are in the `str` type in python3 then you are not supposed to care about encoding at all. That's how it's supposed to be?

Reply:
Pretty much, and the implementation has the freedom to choose the best internal represenation, for python I believe the default is ucs4 but it's configurable at compile time. If I'm not mistaken UCS is used because of it's fixed length, whereas utf8 is variable length and indexing requires a traversal.

Reply:
That used to be the case until Python 3.2 after that they changed he internal representation to always use the most efficient ucs variant based on the largest code point in the particular string: https://www.python.org/dev/peps/pep-0393/